cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -905,19 +905,22 @@ public synchronized void failGeneratingProposalsForExecution(String uuid) {
    *
    * @param forceExecutionStop Whether force execution to stop.
    */
-  public synchronized void userTriggeredStopExecution(boolean forceExecutionStop) {
+  public synchronized void userTriggeredStopExecution(boolean forceExecutionStop, boolean stopExternalAgent) {
     if (stopExecution(forceExecutionStop)) {
       LOG.info("User requested to stop the ongoing proposal execution" + (forceExecutionStop ? " forcefully." : "."));
       _numExecutionStoppedByUser.incrementAndGet();
       _executionStoppedByUser.set(true);
     }
+    if (stopExternalAgent) {
+      maybeStopExternalAgent();
```

Comment:
If this function returns `true`, should we add an info-level log to indicate that the request of the user to stop the execution initiated by an external agent has been successfully submitted?